### PR TITLE
Add API Readiness Checklist content

### LINKED ISSUE #388: Update the API readiness checklist documentation per the new process
**Problem description**
The new release process changes the readiness checklist and the way to handle it. The documentation needs to be updated to reflect this.

**Expected action**
Provide the content of the file ReleaseManagement/documentation/readiness/api-readiness-chacklist.md

**Additional context**
The table of release assets per API status could be added as a checklist to the Release Issue template

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

* documentation

#### What this PR does / why we need it:

Replaces the placeholder in `documentation/readiness/api-readiness-checklist.md` with full content:

- Release assets table (8 items with descriptions, locations, automation status, verification guidance)
- M/O matrix by API status (alpha, rc, initial public, stable public)
- Preparation prerequisites before `/create-snapshot`
- Release readiness review process (codeowner vs release management, permanent vs introduction-phase items)

#### Which issue(s) this PR fixes:

Fixes #388
Supersedes #389

#### Special notes for reviewers:

This content was developed and reviewed as part of [camaraproject/tooling#90](https://github.com/camaraproject/tooling/pull/90) (approved by @tanjadegroot). It is contributed here as the canonical location for the API Readiness Checklist documentation.

### CODE REVIEW

File: documentation/readiness/api-readiness-checklist.md [hunk 1]
```diff
@@ -1,18 +1,111 @@
...
+| Nr | Asset | alpha | rc | initial public | stable public |
+|----|-------|:-----:|:--:|:--------------:|:------------:|
+| 1 | Release Plan | M | M | M | M |
+| 2 | API Definition(s) | M | M | M | M |
+| 3 | Commonalities compliance | O | M | M | M |
+| 4 | API Documentation | M | M | M | M |
+| 5 | User Stories | O | O | O | M |
+| 6 | Test Cases (basic) | O | M | M | M |
+| 7 | Test Cases (enhanced) | O | O | O | M |
+| 8 | API description | O | O | M | M |
```
Comment (tanjadegroot) [inline documentation/readiness/api-readiness-checklist.md]:
```suggestion
| 8 | API Description | O | O | M | M |
```

File: documentation/readiness/api-readiness-checklist.md [hunk 2]
```diff
@@ -1,18 +1,111 @@
...
+| 4 | API Documentation | API description in the YAML `info.description` field; additional documentation as needed | In YAML `info` section or `documentation/` | Partial (`info.description` presence checked) | `info.description` is present and meaningful |
+| 5 | User Stories | At least one user story per API demonstrating the intended use | `documentation/API_documentation/` | No | At least one user story document per API exists |
+| 6 | Test Cases (basic) | Sunny day scenarios and main error cases; at least one `.feature` file per API | `code/Test_definitions/` | File existence only | At least one `.feature` file per API with sunny day + main error cases |
+| 7 | Test Cases (enhanced) | Rainy day scenarios, edge cases, and error handling coverage | `code/Test_definitions/` | No | Additional scenarios beyond basic: rainy day, edge cases, error handling |
+| 8 | API description | Wiki page with content for external visibility | CAMARA Sub Project Wiki | No | Wiki page is present and up to date |
```
Comment (tanjadegroot) [inline documentation/readiness/api-readiness-checklist.md]:
```suggestion
| 8 | API Description | Wiki page with content for external visibility | CAMARA Sub Project Wiki | No | Wiki page is present and up to date |
```

File: documentation/readiness/api-readiness-checklist.md [hunk 3]
```diff
@@ -1,18 +1,111 @@
...
-- [../README.md](../README.md) for documentation index
-- [../deprecated/API-Readiness-Checklist.md](../deprecated/API-Readiness-Checklist.md) for the legacy template
+| Term | Source | Values |
+|------|--------|--------|
+| Release type | `repository.target_release_type` | `pre-release-alpha`, `pre-release-rc`, `public-release`, `maintenance-release` |
+| API status | `apis[].target_api_status` | `alpha`, `rc`, `public` |
+| Initial public | Derived | API status is `public` and major version is 0 (e.g., v0.5.0) |
+| Stable public | Derived | API status is `public` and major version ≥ 1 (e.g., v1.0.0) |
+
+The Release Issue and Release PR embed the readiness matrix as a convenience copy; this document is the canonical source.
```
Comment (tanjadegroot) [inline documentation/readiness/api-readiness-checklist.md]:
```suggestion
The Release Issue and Release PR embed the readiness matrix as a convenience copy; this document holds the reference source.
```
